connmgr/connmanager.go

```diff
@@ -346,6 +352,28 @@ out:
 				log.Debugf("Failed to connect to %v: %v",
 					connReq, msg.err)
 				cm.handleFailedConn(connReq)
+
+			case handleCancelPending:
+				found := false
+				var idToRemove uint64
+				connReq := &ConnReq{}
+				for id, req := range pending {
+					if msg.addr.String() == req.Addr.String() {
+						idToRemove = id
+						connReq = req
+						found = true
+						break
+					}
+
+				}
+				if found {
+					delete(pending, idToRemove)
+					connReq.updateState(ConnCanceled)
+					log.Debugf("Canceled pending connection to %v", msg.addr)
+				} else {
+					log.Errorf("Did not find connection to cancel at address %v", msg.addr)
```

Comment:
This seems like it should be bubbling a not found error back up through the caller so it can report the fact the pending connection was not found as opposed to logging it somewhere the calling RPC client probably won't see.  That would be more consistent with other cases such as where you try to remove a non-permanent peer instead of disconnecting it.